### ISSUE TITLE
azurerm_data_factory Global Parameter not being added

### ISSUE THREAD

**Issue by pimonteiro** [hashicorp/terraform-provider-azurerm#25159]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.4

### AzureRM Provider Version

3.94.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory

### Terraform Configuration Files

```hcl
resource "azurerm_data_factory" "df" {
  name                            = "demo-adf"
  location                        = azurerm_resource_group.rg.location
  resource_group_name             = azurerm_resource_group.rg.name
  managed_virtual_network_enabled = true
  tags = {
    stage   = var.environment
    Project = "DWH"
  }
  dynamic "vsts_configuration" {
    for_each = var.environment == "dev" ? [var.environment] : []

    content {
      account_name    = "account_name"
      branch_name     = "branch_name"
      project_name    = "project_name"
      repository_name = "repo_name"
      root_folder     = var.vstsRootFolder
      tenant_id       = var.tenant-id
    }
  }

  identity {
    type = "SystemAssigned"
  }

  global_parameter {
    name  = "dev_dbricks_clusterID"
    type  = "String"
    value = "123412341234"
  }
  global_parameter {
    name  = "dbr_runtime_version"
    type  = "String"
    value = "12.2.x-scala2.12"
  }
}
```

### Debug Output/Panic Output

```shell
There are many confidential secrets being outputed, it's too much trouble to filter out. And the log is gigantic.
If there's anything I can manually provide, please let me know.
```

### Expected Behaviour

The ADF instance would have an extra **global parameter** added.

### Actual Behaviour

The new **global parameter** does not exist.

### Steps to Reproduce

1. Add global variable to config
2. Run terraform
3. On **terraform plan** confirm that it detects the change
4. Run **terraform apply**
5. Verify in ADF that the variable does not appear

**Comment by SawyerW** [comment]
Yes, I used dynamic feature to create three global parameters and it didn't work.
